azurerm/internal/services/web/resource_arm_app_service_virtual_network_swift_connection_slot.go

```diff
@@ -0,0 +1,216 @@
+	}
+	d.SetId(*read.ID)
+
+	return resourceArmAppServiceVirtualNetworkSwiftConnectionSlotRead(d, meta)
+}
+
+func resourceArmAppServiceVirtualNetworkSwiftConnectionSlotRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["sites"]
+	slotName := id.Path["slots"]
+
+	slot, err := client.GetSlot(ctx, resourceGroup, name, slotName)
+	if err != nil {
+		if utils.ResponseWasNotFound(slot.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving existing App Service Slot %q (App Service %q / Resource Group %q): %s", slotName, name, resourceGroup, err)
+	}
+	appService, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(appService.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving existing App Service %q (Resource Group %q): %s", name, resourceGroup, err)
+	}
+	swiftVnet, err := client.GetSwiftVirtualNetworkConnectionSlot(ctx, resourceGroup, name, slotName)
+	if err != nil {
+		if utils.ResponseWasNotFound(swiftVnet.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving App Service Slot VNet association for %q (App Service %q / Resource Group %q): %s", slotName, name, resourceGroup, err)
+	}
+
+	if swiftVnet.SwiftVirtualNetworkProperties == nil {
+		return fmt.Errorf("Error retrieving virtual network properties (Slot Name %q / App Service %q / Resource Group %q): `properties` was nil", slotName, name, resourceGroup)
+	}
+	props := *swiftVnet.SwiftVirtualNetworkProperties
+	subnetID := props.SubnetResourceID
+	if subnetID == nil || *subnetID == "" {
+		d.SetId("")
+		return nil
+	}
+	d.Set("subnet_id", subnetID)
+	d.Set("app_service_id", appService.ID)
+	d.Set("slot_name", slotName)
+	return nil
+}
+
+func resourceArmAppServiceVirtualNetworkSwiftConnectionSlotDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Get("app_service_id").(string))
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	subnetID, err := azure.ParseAzureResourceID(d.Get("subnet_id").(string))
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", subnetID)
+	}
+	slotName := d.Get("slot_name").(string)
+	resourceGroup := id.ResourceGroup
+	name := id.Path["sites"]
+	subnetName := subnetID.Path["subnets"]
+	virtualNetworkName := subnetID.Path["virtualNetworks"]
```

Comment:
We should be able to use d.ID to delete the resource?

Reply:
I have copied this from the `resource_arm_app_service_virtual_network_switch_connect.go`. But of course. I also changed it in the other file.